src/lucky/page_helpers/url_helpers.cr

```diff
@@ -0,0 +1,65 @@
+module Lucky::UrlHelpers
+  # Tests if the given path matches the current request path.
+  #
+  # ```
+  # # Let's say we are visiting https://example.com/shop/products?order=desc&page=1
+  # current_page?("/shop/checkout")
+  # # => false
+  # current_page?("/shop/products")
+  # # => true
+  # current_page?("/shop/products/")
+  # # => true
+  # current_page?("/shop/products?order=desc&page=1")
+  # # => false
+  # current_page?("/shop/products?order=desc&page=1", check_query_params: true)
+  # # => true
+  # current_page?("https://example.com/shop/products")
+  # # => true
+  # current_page?("https://example.com/shop/products?order=desc&page=1")
+  # # => false
+  # ```
+  def current_page?(
+    path : String,
+    check_query_params : Bool = false
+  )
+    return false unless {"GET", "HEAD"}.includes?(@context.request.method)
+
+    uri = URI.parse(path)
+
+    if check_query_params
+      resource = @context.request.resource
+      path = uri.full_path
+    else
+      resource = URI.parse(@context.request.resource).path
+      path = uri.path
+    end
+
+    unless path == '/'
+      path = path.chomp('/')
+      resource = resource.chomp('/')
+    end
+
+    path == resource
+  end
+
+  # Tests if the given path matches the current request path.
+  #
+  # ```
+  # # Visiting https://example.com/pages/123
+  # current_page?(Pages::Show.with(123))
+  # # => true
+  # current_page?(Pages::Show.with(456))
+  # # => false
+  # # Visiting https://example.com/pages
+  # current_page?(Pages::Index)
+  # # => true
+  # current_page?(Blog::Index)
+  # # => false
+  # ```
+  def current_page?(
+    action : Class | Lucky::RouteHelper,
+    check_query_params : Bool = false
```

Comment:
Is this used?

Reply:
It depends on the question above about `.path`. If path does not take any arguments, this can co.

Reply:
@paulcsmith Can you shed some light here? It's not clear to me from reading the source if for example `Pages::Index.path` or `Pages::Show.with(123).path` would ever need query params as an argument. If not, then we do not need a `check_query_params` argument here. Or am I missing something?

Reply:
What do you get with this?

```
# Visiting https://example.com/action?page=2&per_page=1
view.current_page?("https://example.com/action?page=1&per_page=100")
```

Reply:
That will return `true`, unless you set `check_query_params` to `true`:

```crystal
# Visiting https://example.com/action?page=2&per_page=1
view.current_page?(
  "https://example.com/action?page=1&per_page=100",
  check_query_params: true)
# => false
```

Reply:
But that reminds me of something. This would be `true`:

```crystal
# Visiting https://example.com/action?page=2&per_page=1
view.current_page?("https://example.com/action?page=2&per_page=1")
# => true
```

But this would be `false`:

```crystal
# Visiting https://example.com/action?page=2&per_page=1
view.current_page?("https://example.com/action?per_page=1&page=2")
# => false
```

I'm not sure if that's an issue or not. Will query params always have the same order?

Reply:
No, because we don't know if they were built by hand or not. Like

`a href: "/action?per_page=1&page=#{current_page}"`

Reply:
Ok, in that case the params need to be parsed, sorted and compared.

Reply:
That fixed.